### ISSUE TITLE
{ErrorCannotUseFolderIdForItemId}Expected an item Id but received a folder Id.

### ISSUE THREAD

**Issue by illyax** [ecederstrand/exchangelib#469]
Happens when I create a new Folder instance.

ews_folder = Folder(
    account=account,
    folder_id=folder_info['folder_id'],
    changekey=folder_info['changekey']
)

total_records = ews_folder.all().count()

didn't happened until recently.

**Comment by ecederstrand** [comment]
It looks like a bug in Office365, at least if you're hosted there: https://github.com/OfficeDev/ews-managed-api/issues/204

**Comment by ecederstrand** [comment]
According to the issue I linked to, the Office365 team have fixed the issue now. Can you report back whether the problem is gone on your side?

**Comment by illyax** [comment]
No, it's the same :(

**Comment by ecederstrand** [comment]
Looking at the linked issue again, they state it's being rolled out over the next days. Let's revisit this again later.

**Comment by ecederstrand** [comment]
Closing. Feel free to reopen if this is still a problem.

**Comment by ecboyes** [comment]
@ecederstrand  @illyax This problem is still occurring for me, anyone able to find a solution?

**Comment by ecederstrand** [comment]
@ecboyes Does this mean it hasn't worked since September, or that you just discovered the error? Which Exchange version is this (`print(account.protocol.version)`)? And can you please post your example code?

**Comment by ecboyes** [comment]
The code has been in production since before September and works perfectly fine there. I have read that this does not effect all O365 tenants so I expect that it has something to do with the server in prod vs dev. We are continuing to see the error "The SMTP address has no mailbox associated with it" and have determined that part of the problem is that when Exchange Server 2013 processes a FindItem request, Exchange Server 2013 tries to log the request. That causes an attempt to look at mailbox attributes, but  a mail enabled user does not have the mailbox attributes. Thus Exchange Server 2013 is throwing a "ErrorNonExistentMailbox" error. We were able to "fix" this problem by adding try/catch statements around the "Folder.Bind" call on line 195.  Ultimately we discovered that the root of the problem was at line 95,  "FolderId folderid = new FolderId(folderId)" which is caused by the bug this thread is describing. The error is "Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object". I have attached the code (with a .txt extension instead of .cs extension) for your reference.
[email_copy.txt](https://github.com/ecederstrand/exchangelib/files/3138455/email_copy.txt)
